jsoncdb/org.eclipse.cdt.jsoncdb.core/src/org/eclipse/cdt/jsoncdb/core/internal/builtins/CompilerBuiltinsDetector.java

```diff
@@ -183,19 +184,33 @@ private List<String> getCompilerArguments() {
 	 *         return {@code null}.
 	 */
 	private String[] getEnvp() {
+		var map = new HashMap<String, String>();
+		// The cc.exe from mingw64 (part of Msys2) on Windows needs the bin folder to be on the PATH to be executed.
+		// e.g. 'C:\msys64\mingw64\bin' must be part of the PATH environment variable. That's why we need PATH here:
+		// Fixes CDT #407
+		map.put("PATH", System.getenv("PATH")); //$NON-NLS-1$//$NON-NLS-2$
```

Comment:
The PATH may be modified at CDT build configuration level. I think we need to use CCorePlugin.getDefault().getBuildEnvironmentManager().getVariables(cfg, true) here. Does that make sense?

Reply:
The environment variables of CDT are not used in this case, since it's a a cmake and not a manged build project. There are no env. variables: 
![image](https://github.com/eclipse-cdt/cdt/assets/123444711/044a910b-3365-4330-9f88-656f57179c19)
The `CompilerBuiltinsDetector` is used to parse a compile_commands.json file which is produced by cmake.

Reply:
Hi Gesa, the build environment is available and relevant for any CDT project (not just managed build projects). In the case of tools such as GCC on Windows hosts, CDT detects the location of the tool installation from the Windows Registry and adds the location to the PATH dynamically. If the tool installation is not detected automatically, users may add the tool location to the PATH themselves for relevant build configurations. Typically, the tool location will not be on the global PATH inherited from the operating system.

Reply:
Hi Gesa, I now note from your screenshot that there is no _Environment_ properties page. Is there are reason why this page is supressed? It seems relevant to _all_ CDT projects.

Reply:
Hi John, I don't know. I think the whole cmake stuff was implemented by @15knots .

Reply:
> Hi John, I don't know. I think the whole cmake stuff was implemented by @15knots .

No, it was implemented by Doug Schaefer. I just added the indexer support and a cmake error parser and hooked that in to the existing cmake stuff.